tests/PHPStan/Analyser/nsrt/count-type.php

```diff
@@ -64,6 +64,29 @@ public function doFooBar(
 		}
 	}
 
+	/** @param array{0: string, 1?: string} $arr */
+	public function doBar(array $arr): void
+	{
+		if (count($arr) <= 1) {
+			assertType('1', count($arr));
+			return;
```

Comment:
What about adding `assertType('array{string}', $arr);` ?

Reply:
good point, I already forgot. but quickly looked again and apparently it's currently

```php
assertType('array{0: string, 1?: string}', $arr);
```

it ends up using a falsey scope here, because `count($arr) <= 1` is transformed to `2 < count($arr)` and then something in the loop still skips creating a result type because it can't fully deal with falsey scopes unfortunately. If I remove that `continue`, it would basically come up with "cannot be `array{string, string}`" which should make it then `array{string}` by some falsey scope filtering or so irrelevant of the new code. but that does also not work :/ and many tests for (list) counting start to fail :/ would be great to further improve this, but I was kind of stuck there..